src/inventory_ui.cpp

```diff
@@ -513,15 +516,18 @@ inventory_selector::inventory_selector( const std::string &title, selector_mode
     ctxt.register_action("ANY_INPUT");
 
     player &u = g->u;
-    if (u.is_armed()) {
+    if (u.is_armed() && filter(u.weapon)) {
         worn.push_back(itemstack_or_category(&weapon_cat));
         worn.push_back(itemstack_or_category(&u.weapon, -1));
     }
-    if (!u.worn.empty()) {
-        worn.push_back(itemstack_or_category(&worn_cat));
-    }
     auto iter = u.worn.begin();
     for (size_t i = 0; i < u.worn.size(); i++, ++iter) {
+        if( !filter( *iter ) ) {
+            continue;
+        }
+        if( i == 0 ) {
```

Comment:
This does not work correctly, if the filter does not match the first worn item, but it matches the second one, the category header is not added to the list. Example: wear socks and (loaded) head torch (in that order) and attempt to unload. The game shows the head torch in the worn column but without the ITEMS WORN header. Move the head torch to be the first worn item and it works.